sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java

```diff
@@ -0,0 +1,137 @@
+            .flatMap(byteArrayBody -> {
+                String body = new String(byteArrayBody, StandardCharsets.UTF_8);
+
+                // Split the batch response body into batch operation responses.
+                for (String subResponse : body.split("--" + boundary)) {
+                    // This is a split value that isn't a response.
+                    if (!subResponse.contains("application/http")) {
+                        continue;
+                    }
+
+                    // The batch operation response will be delimited by two new lines.
+                    String[] subResponseSections = subResponse.split(HTTP_NEWLINE + HTTP_NEWLINE);
+
+                    // The first section will contain batching metadata.
+                    BlobBatchOperationResponse<?> batchOperationResponse =
+                        getBatchOperation(batch, subResponseSections[0], logger);
+
+                    // The second section will contain status code and header information.
+                    setStatusCodeAndHeaders(batchOperationResponse, subResponseSections[1]);
+
+                    // The third section will contain the body.
+                    if (subResponseSections.length > 2) {
+                        // The body is optional and may not exist.
+                        setBodyOrPotentiallyThrow(batchOperationResponse, subResponseSections[2], throwOnAnyFailure,
+                            logger);
+                    }
+                }
+
+                return Mono.just(new SimpleResponse<>(batchResponse, null));
+            });
+    }
+
+    private static BlobBatchOperationResponse<?> getBatchOperation(BlobBatch batch, String responseBatchInfo,
+        ClientLogger logger) {
+        Matcher contentIdMatcher = CONTENT_ID_PATTERN.matcher(responseBatchInfo);
+
+        int contentId;
+        if (contentIdMatcher.find()) {
+            contentId = Integer.parseInt(contentIdMatcher.group(1));
+        } else {
+            throw logger.logExceptionAsError(
+                new IllegalStateException("Batch operation response doesn't contain a 'Content-Id' header."));
+        }
+
+        return batch.getBatchRequest(contentId).setResponseReceived();
+    }
+
+    private static void setStatusCodeAndHeaders(BlobBatchOperationResponse<?> batchOperationResponse,
+        String responseHeaders) {
+        HttpHeaders headers = new HttpHeaders();
+        for (String line : responseHeaders.split(HTTP_NEWLINE)) {
+            if (ImplUtils.isNullOrEmpty(line)) {
+                continue;
+            }
+
+            if (line.startsWith("HTTP")) {
+                Matcher statusCodeMatcher = STATUS_CODE_PATTERN.matcher(line);
+                if (statusCodeMatcher.find()) {
+                    batchOperationResponse.setStatusCode(Integer.parseInt(statusCodeMatcher.group(1)));
+                }
+            } else {
+                String[] headerPieces = line.split(":\\s*", 2);
```

Comment:
If the line doesn't contain that split `":\\s*"`, it'll only result in a single array item, causing your following code "headerPieces[1]" to fail with an ArrayOutOfBounds exception.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/5756